ADR/0009-pipeline-service-via-operator.md

```diff
@@ -0,0 +1,70 @@
+cluster (specifically OpenShift Dedicated).
+
+## Decision
+
+All Tekton APIs will be provided using the stock OpenShift Pipelines operator (OSP).
+In the spirit of developing in a "Service First" manner, RHTAP will deploy
+a candidate "nightly" release of the operator. The service will be defined in
+the [pipeline-service](https://github.com/openshift/pipeline-service)
+repository, which is then imported into
+[infra-deployments](https://github.com/redhat-appstudio/infra-deployments) as
+an ArgoCD application.
+
+Configurations that are specific to RHTAP must be made available through
+the OSP CRDs. The following changes are specific to RHTAP:
+
+- Disable Tekton Triggers. Pipelines as Code will be the favored mechanism for
+  event-based triggering of pipelines for now. This decision can be revisited
+  in the future based on need.
+- Disable the pruner that comes with the operator. Tekton Results will be used
+  to prune `PipelineRun` and `TaskRun` data off cluster, thereby ensuring data
+  is archived before removal.
+- Direct Pipelines as Code to use a URL pattern that displays the `PipelineRun`
+  or `TaskRun` info from the Hybrid Application Console (HAC). This ensures
+  end users do not need access to the underlying compute cluster(s).
+- The Pipelines as Code application name must match the GitHub Application name.
+- The GitHub Application secret.
+- Any configuration related to performance.
+
+Furthermore, as the service will be accessed through CodeReadyToolchain (CRT), the
+following changes are also specific to RHTAP:
+- Deploying a proxy (known as `SprayProxy`) on the CRT host cluster that redirects
+  incoming PaC requests to the member clusters. Requests are redirected only if:
+  - the webhook secret is valid;
+  - the request size is under 25MB.
+- Providing a plugin to the CRT Proxy so Tetkon Result requests are redirected
+  to the appropriate member cluster.
+
+## Consequences
+
+- Tekton Triggers should be disabled in RHTAP using the appropriate operator
+  configuration.
+- The Tekton Pruner needs to be disabled in RHTAP using the appropriate
+  operator configuration. This is done under the assumption that Results will
+  be responsible for pruning resources. Eventually the operator should automate
+  this setting if `Results` is deployed and configured to prune resources.
+- Pipelines as Code should use an appropriate URL to HAC when interacting with
+  SCM services, such as the GitHub
+  [Checks API](https://docs.github.com/en/rest/guides/getting-started-with-the-checks-api?apiVersion=2022-11-28).
+- Changes to Pipeline components need to be baked into the operator and built
+  rapidly.
+- Hot fixes need to be provided on a "roll forward" basis. OLM Operators do not
+  support "rollback" mechanisms today. Mean time to revert an offending change,
+  rebuild, and deploy needs to be measured in hours.
+- The version of the deployed operator needs to be configurable via ArgoCD.
```

Comment:
We only change the catalog source, specifically the image. The new image is expected to include a newer version of the OSP operator. We also update the ImageContentSourcePolicy where the images required by the operator are pulled from.